src/java/org/pantsbuild/tools/junit/impl/ConsoleRunnerImpl.java

```diff
@@ -484,11 +526,33 @@ private Thread createUnexpectedExitHook(final ShutdownListener listener, final P
         // not want to go unnoticed.
         out.println("FATAL: VM exiting unexpectedly.");
         out.flush();
-        Runtime.getRuntime().halt(1);
+        try {
+          // Other shutdown hooks might still need to write test results,
+          // so wait for them (up to 2 seconds) to finish before halting.
+          haltAfter.countDown();
+          haltAfter.await(2, TimeUnit.SECONDS);
```

Comment:
`2` feels short enough to be potentially flaky on a busy host. Would wait a bit longer (`10`, `15` ish) unless we're expecting those threads to actually be dead fairly frequently.